pandas/core/strings.py

```diff
@@ -372,27 +375,51 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
     0    tWO
     1    bAR
     dtype: object
+
+    When `pat` is a compiled regex, all flags should be included in the
+    compiled regex. `case` and `flags` are otherwise ignored.
+
+    >>> regex_pat = re.compile(r'FUZ', flags=re.IGNORECASE)
+    >>> pd.Series(['foo', 'fuz', np.nan]).str.replace(regex_pat, 'bar')
+    0    foo
+    1    bar
+    2    NaN
+    dtype: object
     """
 
     # Check whether repl is valid (GH 13438, GH 15055)
     if not (is_string_like(repl) or callable(repl)):
         raise TypeError("repl must be a string or callable")
-    use_re = not case or len(pat) > 1 or flags or callable(repl)
-
-    if use_re:
+    # Check whether pat is a compiled regex or should be compiled
+    is_re = isinstance(pat, type(re.compile('')))
+    build_re = not is_re and (
+        not case or len(pat) > 1 or flags or callable(repl))
+
+    if is_re:
+        if not case or flags:
+            warnings.warn(
```

Comment:
this is a problem.  If you want to ignore flags/case, then you need to change those defaults to ``None``. Then if you have a compilable you can check this (and raise, not warn) if the are not None. If you have a string, then these can be None, at which point you set the default. This should be back-compat.